service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -301,45 +326,91 @@ public void addComplyOnBehalfResponsesToDirectionsInStandardDirectionsOrder(Case
                         .map(response -> response.toBuilder().respondingOnBehalfOf("CAFCASS").build())
                         .collect(toList());
 
-                    addResponsesToDirections(responses, caseData.getStandardDirectionOrder().getDirections());
+                    addResponsesToDirections(responses, directionsToComplyWith);
 
                     break;
+
                 case PARENTS_AND_RESPONDENTS:
+                    List<Element<DirectionResponse>> respondentResponses = addValuesToListResponses(
+                        directions, eventId, authorisation, PARENTS_AND_RESPONDENTS);
+
+                    addResponseElementsToDirections(respondentResponses, directionsToComplyWith);
+
+                    break;
                 case OTHERS:
-                    List<Element<DirectionResponse>> elements = addValuesToListResponseDirections(directions);
+                    List<Element<DirectionResponse>> otherResponses = addValuesToListResponses(
+                        directions, eventId, authorisation, OTHERS);
 
-                    addResponseElementsToDirections(elements, caseData.getStandardDirectionOrder().getDirections());
+                    addResponseElementsToDirections(otherResponses, directionsToComplyWith);
 
```

Comment:
Are we doing anything different in that block? Maybe that should be default? or two scenarios passing the assignee value?

Reply:
hmm, so what we don't want to happen here is act upon other values for assignee... default would catch these other values hence them being split. Any ideas on how to improve?

Reply:
I was simply thinking about doing:
```
                case PARENTS_AND_RESPONDENTS:
                case OTHERS:
                    directions.addAll(clone);

                    if (eventId == COMPLY_ON_BEHALF_SDO) {

but there is string value different than the enum, so happy for now to leave it as it is 👍